artemis-server/src/test/java/org/apache/activemq/artemis/core/server/files/FileStoreMonitorTest.java

```diff
@@ -72,71 +72,117 @@ public void testSimpleTick() throws Exception {
 
       bufferedOutputStream.close();
 
-      final AtomicInteger over = new AtomicInteger(0);
-      final AtomicInteger under = new AtomicInteger(0);
+      final AtomicInteger overMaxUsage = new AtomicInteger(0);
+      final AtomicInteger underMaxUsage = new AtomicInteger(0);
       final AtomicInteger tick = new AtomicInteger(0);
 
-      FileStoreMonitor.Callback callback = new FileStoreMonitor.Callback() {
-         @Override
-         public void tick(long usableSpace, long totalSpace) {
-            tick.incrementAndGet();
-            logger.debug("tick:: usableSpace: {}, totalSpace:{}", usableSpace, totalSpace);
+      FileStoreMonitor.Callback callback = (usableSpace, totalSpace, ok, type) -> {
+         tick.incrementAndGet();
+         if (ok) {
+            underMaxUsage.incrementAndGet();
+         } else {
+            overMaxUsage.incrementAndGet();
          }
+         logger.debug("tick:: usableSpace: " + usableSpace + ", totalSpace:" + totalSpace);
+      };
+      FileStoreMonitor storeMonitor = new FileStoreMonitor(scheduledExecutorService, executorService, 100L, TimeUnit.MILLISECONDS, 0.999, null, FileStoreMonitor.FileStoreMonitorType.MaxDiskUsage);
+      storeMonitor.addCallback(callback);
+      storeMonitor.addStore(getTestDirfile());
 
-         @Override
-         public void over(long usableSpace, long totalSpace) {
-            over.incrementAndGet();
-            logger.debug("over:: usableSpace: {}, totalSpace:{}", usableSpace, totalSpace);
-         }
+      storeMonitor.tick();
+
+      Assert.assertEquals(0, overMaxUsage.get());
+      Assert.assertEquals(1, tick.get());
+      Assert.assertEquals(1, underMaxUsage.get());
 
-         @Override
-         public void under(long usableSpace, long totalSpace) {
-            under.incrementAndGet();
-            logger.debug("under:: usableSpace: {}, totalSpace: {}", usableSpace, totalSpace);
+      storeMonitor.setMaxUsage(0);
+
+      storeMonitor.tick();
+
+      Assert.assertEquals(1, overMaxUsage.get());
+      Assert.assertEquals(2, tick.get());
+      Assert.assertEquals(1, underMaxUsage.get());
+   }
+
+   @Test
+   public void testSimpleTickForMinDiskFree() throws Exception {
+      File garbageFile = new File(getTestDirfile(), "garbage.bin");
+      FileOutputStream garbage = new FileOutputStream(garbageFile);
+      BufferedOutputStream bufferedOutputStream = new BufferedOutputStream(garbage);
+      PrintStream out = new PrintStream(bufferedOutputStream);
+
+      // This is just to make sure there is at least something on the device.
+      // If the testsuite is running with an empty tempFS, it would return 0 and the assertion would fail.
+      for (int i = 0; i < 100; i++) {
+         out.println("Garbage " + i);
+      }
+
+      bufferedOutputStream.close();
```

Comment:
Could this use try-with-resources to eliminate the need to explicitly close, and make it safer?